### PR TITLE
Enable linker warnings as error for Windows

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Enabled linker warnings as error for Windows

### CODE REVIEW

File: src/inference/include/ie/ie_common.h [hunk 1]
```diff
@@ -384,15 +384,14 @@ namespace details {
...
 /**
  * @brief Tag struct used to throw exception
  */
 template <typename ExceptionType>
 struct ThrowNow final {
     [[noreturn]] void operator<<=(const std::ostream& ostream) {
-        std::ostringstream stream;
-        stream << ostream.rdbuf();
-        throw ExceptionType{stream.str()};
+        throw ExceptionType{get_message(ostream)};
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/ie_common.h:394]: what's problem here?

Comment (ilyachur) [reply src/inference/include/ie/ie_common.h:394]: I will revert this change. It was needed in order to understand the issue

File: src/plugins/intel_gpu/src/kernel_selector/CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +59,10 @@ add_custom_command(OUTPUT "${CODEGEN_INCDIR}/${PRIM_DB}"
 
 add_library(${TARGET_NAME} STATIC ${LIBRARY_SRC} ${CODEGEN_CACHE_SOURCES})
 
+if(NOT BUILD_SHARED_LIBS)
+    target_compile_definitions(${TARGET_NAME} PUBLIC OPENVINO_STATIC_LIBRARY)
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/kernel_selector/CMakeLists.txt:63]: generator expression?

File: src/frontends/tests/frontend/shared/test_builtin_extensions_2/CMakeLists.txt [hunk 3]
```diff
@@ -4,6 +4,12 @@
 
 set(TARGET_NAME "test_builtin_extensions_2")
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /ignore:4217")
+    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} /ignore:4217")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /ignore:4217")
```
Comment (ilya-lavrenov) [inline src/frontends/tests/frontend/shared/test_builtin_extensions_2/CMakeLists.txt:10]: can we add comments why we disable such warnings? Is it a temporary solution?

Comment (ilyachur) [reply src/frontends/tests/frontend/shared/test_builtin_extensions_2/CMakeLists.txt:10]: Yes I will create tickets to solve all warnings.

File: src/plugins/intel_gpu/src/kernel_selector/CMakeLists.txt (hunk 2 above)
Comment (ilyachur) [reply src/plugins/intel_gpu/src/kernel_selector/CMakeLists.txt:63]: I try to fix windows static build (on my local machine all passed :) )

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**